share/CMakeJS.cmake

```
@@ -0,0 +1,642 @@
+#[=============================================================================[
+  CMakeJS.cmake - A proposed CMake API for cmake-js v8
+  Copyright (C) 2024 Nathan J. Hood
+  MIT License
+  See: https://github.com/nathanjhood/NapiAddon
+]=============================================================================]#
+
+#[=============================================================================[
+Check whether we have already been included (borrowed from CMakeRC)
+]=============================================================================]#
+# TODO: Decouple CMakeJS.cmake API version number from cmake-js version number...?
+set(_version 7.3.3)
+
```

```suggestion

if (DEFINED CMAKE_JS_VERSION)
    message(FATAL_ERROR "You cannot use this new cmake flow with the old cmake-js binary, instead you should use cmake-js2 or cmake")
endif()

```

While this adds some pain for now, I don't want this being used through the current `cmake-js` binary, as the variables and defines it injects will cause things to break once those get removed.